frontend/controller/actions/group.js

```diff
@@ -914,41 +912,92 @@ export default (sbp('sbp/selectors/register', {
   ...encryptedAction('gi.actions/group/paymentUpdate', L('Failed to update payment.')),
   ...encryptedAction('gi.actions/group/sendPaymentThankYou', L('Failed to send a payment thank you note.')),
   ...encryptedAction('gi.actions/group/groupProfileUpdate', L('Failed to update group profile.')),
-  ...encryptedAction('gi.actions/group/proposal', L('Failed to create proposal.')),
+  ...encryptedAction('gi.actions/group/proposal', L('Failed to create proposal.'), async (sendMessage, params) => {
+    const { contractID } = params
+    await sendMessage({
+      ...params,
+      hooks: {
+        onprocessed: (message) => {
+          (async () => {
+            const proposalId = message.hash()
+            const state = await sbp('chelonia/contract/state', contractID)
+            const proposal = state.proposals[proposalId]
+            const proposalToSend = { ...extractProposalData(proposal, { proposalId }), status: STATUS_OPEN }
+
+            await sbp('gi.actions/group/notifyProposalStateInGeneralChatRoom', { groupID: contractID, proposal: proposalToSend })
+          })()
+        }
```

Comment:
You don't need to create an async function inside of `onprocessed`. You can write this like so:

```js
onprocessed: async (message) => {
  // ...
```

This also should be rewritten:

```js
const proposalToSend = { ...extractProposalData(proposal, { proposalId }), status: STATUS_OPEN }
```

Like this:

```js
const proposalToSend = extractProposalData(proposal, { proposalId, status: STATUS_OPEN })
```

Finally, you should probably wrap the contents of the event handler in a try/catch block and `console.error` if there's an error.

Reply:
```
export type GIActionParams = {
  action: string;
  contractID: string;
  data: Object;
  options?: Object; // these are options for the action wrapper
  signingKeyId: ?string;
  encryptionKeyId: ?string;
  originatingContractID: ?string;
  signingContractID: ?string;
  innerSigningContractID: ?string;
  innerSigningKeyId: ?string;
  hooks?: {
    preSendCheck?: null | (Object, Object) => boolean;
    prepublishContract?: null | (Object) => void;
    prepublish?: null | (Object) => void;
    postpublish?: null | (Object) => void;
    onprocessed?: null | (Object) => void;
  };
  publishOptions?: { maxAttempts: number };
  returnInvocation?: boolean;
}
```
Since the `onprocessed` hook function is not defined as to return `Promise`, I couldn't do that way.
Also, since `onprocessed`, `prepublish`, `postpublish` are all defined to return `void`, not `Promise` in chelonia, if I change this `GIActionParams`, I also need to change many types too; `ChelRegParams`, `ChelActionParams`, `ChelKeyAddParams`, ...
However, I can see that the in the usage of those hooks, we assumed they would be async functions. Yeah, it's confusion. See the code below. 
```
await hooks?.postpublish?.(entry)
```
That's why I made that way. However, I will add `status` value inside.

Reply:
If we do `await hooks?.postpublish?.(entry)` then it is correct to define them as returning a promise.

So feel free to update `(Object) => void` to `(Object) => Promise<void>;`

Reply:
Okay. I will update all the types of GIActionParams and types of Chelonia too.